executor/cook/executor.py

```diff
@@ -0,0 +1,363 @@
+        # updater termination if executor is terminating normally
+        completed_signal.set()
+        if not stop_signal.isSet():
+            logging.info('Awaiting progress updater completion with timeout of one second')
+            progress_complete_event.wait(1)
+
+        logging.info('Process completed with exit code {}'.format(exit_code))
+        exit_message = json.dumps({'exit-code': exit_code, 'task-id': task_id})
+        send_message(driver, exit_message, config.max_message_length)
+
+        # send the latest progress state if available
+        cp.force_send_progress_update(progress_watcher, progress_updater)
+
+        # task either completed successfully or aborted with an error
+        task_state = get_task_state(exit_code)
+        update_status(driver, task_id, task_state)
+
+    except Exception:
+        # task aborted with an error
+        logging.exception('Error in executing task')
+        update_status(driver, task_id, cook.TASK_FAILED)
+
+    finally:
+        # ensure completed_signal is set
+        completed_signal.set()
+
+
+def run_mesos_driver(stop_signal, config):
+    """Run an executor driver until the stop_signal event is set or the first task it runs completes."""
+    executor = CookExecutor(stop_signal, config)
+    driver = MesosExecutorDriver(executor)
+    driver.start()
+
+    # check the status of the executor and bail if it has crashed
+    while not executor.has_stopped():
+        time.sleep(1)
+    else:
+        logging.info('Executor thread has completed')
+
+    driver.stop()
+
+
+class CookExecutor(Executor):
+    """This class is responsible for launching the task sent by the scheduler.
+    It implements the Executor methods."""
+
+    def __init__(self, stop_signal, config):
+        self.stop_signal = stop_signal
+        self.completed_signal = Event()
+        self.config = config
+
+    def registered(self, driver, executor_info, framework_info, agent_info):
+        logging.info('Executor registered executor={}, framework={}, agent={}'.
+                     format(executor_info['executor_id']['value'], framework_info['id'], agent_info['id']['value']))
+
+    def reregistered(self, driver, agent_info):
+        logging.info('Executor re-registered agent={}'.format(agent_info))
+
+    def disconnected(self, driver):
+        logging.info('Executor disconnected!')
+
+    def launchTask(self, driver, task):
+        logging.info('Driver {} launching task {}'.format(driver, task))
+        thread = Thread(target=manage_task,
+                        args=(driver, task, self.stop_signal, self.completed_signal, self.config, 'stdout', 'stderr'))
+        thread.start()
+
+    def killTask(self, driver, task_id):
+        logging.info('Task {} has been killed by Mesos'.format(task_id))
+        self.stop_signal.set()
+
+    def shutdown(self, driver):
+        logging.info('Mesos requested executor to shutdown!')
+        self.stop_signal.set()
+
+    def has_stopped(self):
```

Comment:
The other methods override `Executor` methods which use camelcase.

Will rename using underscores.